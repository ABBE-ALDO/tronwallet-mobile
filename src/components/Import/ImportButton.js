import React from 'react'
import { ActivityIndicator } from 'react-native'
import { func, bool } from 'prop-types'

import tl from '../../utils/i18n'
import { Colors } from '../DesignSystem'
import ButtonGradient from '../ButtonGradient'

const ImportButton = ({ loading, onPress, disabled }) =>
  loading
    ? (<ActivityIndicator size='small' color={Colors.primaryText} />)
    : (
      <ButtonGradient
        font='bold'
        text={tl.t('importWallet.button')}
        onPress={onPress}
        disabled={disabled}
      />
    )

ImportButton.propTypes = {
  loading: bool,
  onPress: func.isRequired,
  disabled: bool
}

ImportButton.defaultProps = {
  loading: false,
  disabled: false
}

export default ImportButton
