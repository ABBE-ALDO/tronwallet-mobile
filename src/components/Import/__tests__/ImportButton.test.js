import React from 'react'
import { shallow } from 'enzyme'
import { ActivityIndicator } from 'react-native'

import ImportButton from '../ImportButton'

jest.mock('../../../utils/i18n')

describe('Import button component', () => {
  const setup = propOverrides => {
    const props = Object.assign({
      onPress: () => {}
    }, propOverrides)

    const wrapper = shallow(
      <ImportButton {...props} />
    )

    return {
      props,
      wrapper
    }
  }

  test('Should match when render component', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  test('Should not loading when loading props is false by default', () => {
    const { wrapper } = setup()
    expect(wrapper.find(ActivityIndicator).exists()).toBe(false)
  })

  test('Should loading when loading props is true', () => {
    const { wrapper } = setup({ loading: true })
    expect(wrapper.find(ActivityIndicator).exists()).toBe(true)
  })
})
