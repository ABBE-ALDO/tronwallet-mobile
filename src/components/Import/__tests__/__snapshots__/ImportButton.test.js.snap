// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Import button component Should match when render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ImportButton
    disabled={false}
    loading={false}
    onPress={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "disabled": false,
      "font": "bold",
      "leftRadius": 4,
      "onPress": [Function],
      "rightRadius": 4,
      "size": "large",
      "text": "IMPORT",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "disabled": false,
        "font": "bold",
        "leftRadius": 4,
        "onPress": [Function],
        "rightRadius": 4,
        "size": "large",
        "text": "IMPORT",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenThreeAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
